js/src/components/paid-ads/constants.js

```diff
@@ -0,0 +1,282 @@
+		heading: _x(
+			'Descriptions',
+			'Plural asset field name as the heading',
+			'google-listings-and-ads'
+		),
+		addButtonText: __( 'Add description', 'google-listings-and-ads' ),
+		capitalizedName: _x(
+			'Description',
+			'Capitalized asset field name as the placeholder or the start of an error message',
+			'google-listings-and-ads'
+		),
+		lowercaseSingularName: _x(
+			'description',
+			'Singular and lowercase asset field name',
+			'google-listings-and-ads'
+		),
+		lowercasePluralName: _x(
+			'descriptions',
+			'Plural and lowercase asset field name',
+			'google-listings-and-ads'
+		),
+		help: (
+			<>
+				<p>
+					{ __(
+						'The description adds to the headline and provides additional context or details. It can be up to 90 characters, and may appear after the headline.',
+						'google-listings-and-ads'
+					) }
+				</p>
+				<p>
+					{ __(
+						`The length of the rendered description will depend on the site it appears on. If it's shortened, it will end with an ellipsis(…). The description doesn't show in all sizes and formats.`,
+						'google-listings-and-ads'
+					) }
+				</p>
+			</>
+		),
+	},
+];
+
+{
+	function getSubheading( min, max ) {
+		return sprintf(
+			// translators: 1: The minimal number of this item. 2: The maximum number of this item.
+			__(
+				'At least %1$d required. Add up to %2$d.',
+				'google-listings-and-ads'
+			),
+			min,
+			max
+		);
+	}
+
+	function getImageHelpContent( subheading, imageConfig ) {
+		const size = sprintf(
+			// translators: 1: Recommended width. 2: Recommended height. 3: Minimal width. 4: Minimal height.
+			__(
+				'Recommended size: %1$d x %2$d<newline />Min. size: %3$d x %4$d',
+				'google-listings-and-ads'
+			),
+			imageConfig.suggestedWidth,
+			imageConfig.suggestedHeight,
+			imageConfig.minWidth,
+			imageConfig.minHeight
+		);
+		return (
+			<>
+				<p>
+					{ __(
+						'Add images that meet or can be cropped to the recommended sizes. Note: The maximum file size for any image is 5120 KB.',
+						'google-listings-and-ads'
+					) }
+				</p>
+				<p>
+					<strong>{ subheading }</strong>
+				</p>
+				<p>{ createInterpolateElement( size, { newline: <br /> } ) }</p>
+			</>
+		);
+	}
+
+	ASSET_IMAGE_SPECS.forEach( ( spec ) => {
+		spec.subheading = getSubheading( spec.min, spec.max );
+		spec.help = getImageHelpContent(
+			spec.helpSubheading,
+			spec.imageConfig
+		);
+	} );
+
+	ASSET_TEXT_SPECS.forEach( ( spec ) => {
+		spec.subheading = getSubheading( spec.min, spec.max );
+	} );
+}
+
+const ASSET_DISPLAY_URL_PATH_SPECS = [
+	{
+		maxCharacterCount: 15,
+		capitalizedName: _x(
+			'The first display URL path',
+			'Capitalized asset field name as the start of an error message',
+			'google-listings-and-ads'
+		),
+	},
+	{
+		maxCharacterCount: 15,
+		capitalizedName: _x(
+			'The second display URL path',
+			'Capitalized asset field name as the start of an error message',
+			'google-listings-and-ads'
+		),
+	},
+];
```

Comment:
💅 WDYT about moving this one at the top to avoid the dynamic logic and the private functions `getImageHelpContent` and `getSubheading` to be mixed between the constants.

Reply:
Moved by https://github.com/woocommerce/google-listings-and-ads/pull/1863/commits/03dd7667ef2c109b7490ba83ca15b482f1afc49a.